### ISSUE TITLE
List all the App Service Slots for a given App Service Plan

### ISSUE THREAD

**Issue by ravitellakula** [Azure/azure-cli#23924]
**Related command**

`az webapp list --query "[?appServicePlanId=='$appServicePlanID']"`

**Is your feature request related to a problem? Please describe.**

I do have a requirement to change the vnet integration for all my applications to move to a new subnet. And, I was hoping to list all the web apps (app services) and corresponding slots in an app service plan - run through the list iteratively to remove the vnet integration and then associate it back. 

**Describe the solution you'd like**

Similar to the command posted above, where I can list all the app services in a given plan - would like to have the same feature for `az webapp deployment slot`

**Describe alternatives you've considered**

I was trying my luck with `az resource list`, while I can list my app service and their corresponding slots with this command I cannot restrict it to a specific app service plan.

`az resource list --resource-group $resourceGroupName --query "[?kind=='app,linux' && (type=='Microsoft.Web/sites' || type=='Microsoft.Web/sites/slots')]"`

**Additional context**

**Comment by yonzhan** [comment]
route to CXP team

**Comment by SaurabhSharma-MSFT** [comment]
@ravitellakula I do not think we have any CLI to get the same. You can probably call the below REST API to get apps and respective slots under an app service plan- `https://management.azure.com//subscriptions/{subscriptionid}/resourceGroups/{resourcegroup}/providers/Microsoft.Web/serverfarms/{AppServicePlanName}/sites?includeSlots=true`.

**Comment by seligj95** [comment]
> [!NOTE]
> **[AI-Assisted Triage]** This comment was generated with AI assistance as part of an App Service CLI issue review. Reply to this comment and tag @seligj95 to follow up.

**Summary**: Request to list deployment slots associated with a given App Service Plan — currently there's no easy way to see which slots are consuming plan resources.

**Status**: Feature Approved

**Priority**: P3

**Next Steps**: Feature enhancement planned — adding the ability to list slots for a given App Service Plan.
